ci-operator/step-registry/azure/cluster/custom-dns/azure-cluster-custom-dns-commands.sh

```diff
@@ -0,0 +1,131 @@
+        export REQUESTS_CA_BUNDLE=/tmp/ca.pem
+    fi
+    az cloud register \
+        -n ${cloud_name} \
+        --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+        --suffix-storage-endpoint "${SUFFIX_ENDPOINT}"
+    az cloud set --name ${cloud_name}
+    az cloud update --profile 2019-03-01-hybrid
+else
+    az cloud set --name AzureCloud
+fi
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+az account set --subscription ${AZURE_AUTH_SUBSCRIPTION_ID}
+
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+INFRA_ID=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+CLUSTER_NAME=$(yq-go r "${INSTALL_CONFIG}" 'metadata.name')
+RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.resourceGroupName')
+if [[ -z "${RESOURCE_GROUP}" ]]; then
+    RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+
+api_dns="api.${CLUSTER_NAME}.${BASE_DOMAIN}"
+ingress_dns="*.apps.${CLUSTER_NAME}.${BASE_DOMAIN}"
+customer_dns_json="[]"
+
+PUBLISH=$(yq-go r "${INSTALL_CONFIG}" 'publish')
+PUBLISH="${PUBLISH:-External}"
+apiserver_publish="${PUBLISH}"
+ingress_publish="${PUBLISH}"
+
+if [[ "${PUBLISH}" == "Mixed" ]]; then
+    apiserver_config="$(yq-go r "${INSTALL_CONFIG}" 'operatorPublishingStrategy.apiserver')"
+    apiserver_publish=${apiserver_config:-"External"}
+    ingress_config="$(yq-go r "${INSTALL_CONFIG}" 'operatorPublishingStrategy.ingress')"
+    ingress_publish=${ingress_config:-"External"}
+fi
+
+
+api_lb_ip=""
+if [[ "${apiserver_publish}" == "External" ]]; then
+    api_frontendipconfig_id=$(az network lb show -n ${INFRA_ID} -g ${RESOURCE_GROUP} -ojson | jq -r ".loadBalancingRules[] | select(.frontendPort == 6443) | .frontendIPConfiguration.id")
+    api_frontendipconfig_name=${api_frontendipconfig_id##*/}
+    api_frontendpublicip_id=$(az network lb frontend-ip show -n ${api_frontendipconfig_name} --lb-name ${INFRA_ID} -g ${RESOURCE_GROUP} --query "publicIPAddress.id" -otsv)
+    api_lb_ip=$(az network public-ip show --ids ${api_frontendpublicip_id} --query 'ipAddress' -otsv)
+    echo "apiserver rule's frontend public IP in public LB: ${api_lb_ip}"
+elif [[ "${apiserver_publish}" == "Internal" ]]; then
+    api_frontendipconfig_id=$(az network lb show -n ${INFRA_ID}-internal -g ${RESOURCE_GROUP} -ojson | jq -r ".loadBalancingRules[] | select(.frontendPort == 6443) | .frontendIPConfiguration.id")
+    api_frontendipconfig_name=${api_frontendipconfig_id##*/}
+    api_lb_ip=$(az network lb frontend-ip show -n ${api_frontendipconfig_name} --lb-name ${INFRA_ID}-internal -g ${RESOURCE_GROUP} --query "privateIPAddress" -otsv)
+    echo "apiserver rule's fronend private IP in internal LB: ${api_lb_ip}"
+fi
+
+if [[ -n "${api_lb_ip}" ]]; then
+    customer_dns_json=$(echo "${customer_dns_json}" | jq -c -S ". +=[{\"name\":\"${api_dns}\",\"target\":\"${api_lb_ip}\",\"record_type\":\"A\"}]")
```

Comment:
How about introduce the same [function](https://github.com/openshift/release/pull/66644/files#diff-0f4c428eeae557540ef010250dde7ede27803cca9ee8f7426d02d696e71424a7R20) for json write operation ?

Reply:
updated.